packages/commercetools/theme/pages/MyAccount/OrderHistory.vue

```diff
@@ -0,0 +1,287 @@
+            <SfTableHeader
+              v-for="tableHeader in tableHeaders"
+              :key="tableHeader"
+            >{{ tableHeader }}</SfTableHeader>
+            <SfTableHeader class="orders__element--right" />
+          </SfTableHeading>
+          <SfTableRow v-for="order in orders" :key="orderGetters.getId(order)">
+            <SfTableData v-e2e="'order-number'">{{ orderGetters.getId(order) }}</SfTableData>
+            <SfTableData>{{ orderGetters.getDate(order) }}</SfTableData>
+            <SfTableData>{{ $n(orderGetters.getPrice(order), 'currency') }}</SfTableData>
+            <SfTableData>
+              <span :class="getStatusTextClass(order)">{{ orderGetters.getStatus(order) }}</span>
+            </SfTableData>
+            <SfTableData class="orders__view orders__element--right">
+              <SfButton class="sf-button--text desktop-only" @click="currentOrder = order">
+                {{ $t('View details') }}
+              </SfButton>
+            </SfTableData>
+          </SfTableRow>
+        </SfTable>
+        <div class="pagination" v-if="orders.length < totalOrders">
+          <SfArrow
+            aria-label="prev"
+            class="sf-arrow--left sf-arrow--transparent"
+            :disabled="offset === 0"
+            @click="goPrev(offset)"
+          />
+          <div class="pagination-count">
+            {{ orders.length > 1 ? `${offset + 1} - ` : "" }} {{ offset + orders.length }}
+            <strong>of</strong> {{ totalOrders }}
+          </div>
+          <SfArrow
+            aria-label="next"
+            class="sf-arrow--right sf-arrow--transparent"
+            :disabled="(offset + orders.length) === totalOrders"
+            @click="goNext(offset)"
+          />
+        </div>
+      </div>
+    </SfTab>
+    <SfTab title="Returns">
+      <p class="message">
+        This feature is not implemented yet! Please take a look at
+        <br />
+        <SfLink class="message__link" href="#">https://github.com/DivanteLtd/vue-storefront/issues</SfLink>
+        for our Roadmap!
+      </p>
+    </SfTab>
+  </SfTabs>
+</template>
+
+<script>
+import {
+  SfTabs,
+  SfTable,
+  SfButton,
+  SfProperty,
+  SfLink,
+  SfArrow
+} from '@storefront-ui/vue';
+import { computed, ref } from '@vue/composition-api';
+import { useUserOrder, orderGetters } from '@vue-storefront/commercetools';
+import { AgnosticOrderStatus } from '@vue-storefront/core';
+import { onSSR } from '@vue-storefront/core';
+
+export default {
+  name: 'PersonalDetails',
+  components: {
+    SfTabs,
+    SfTable,
+    SfButton,
+    SfProperty,
+    SfLink,
+    SfArrow
+  },
+  setup() {
+    const limit = 10;
+    const { orders, search } = useUserOrder();
+    const currentOrder = ref(null);
+
+    onSSR(async () => {
+      await search({ limit, offset: 0, sort: 'createdAt desc' });
+    });
+
+    const goNext = (offset) => {
+      search({ limit, offset: offset + limit, sort: 'createdAt desc' });
```

Comment:
Didn't want you to create single method that'll accept direction (prev/next) as a param to search for items? Then you could use it here, in `goPrev` and on init